app/models/category.py

```diff
@@ -1,102 +1,122 @@
-        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
-    )
-
-    total_goal: Mapped[int] = mapped_column(Integer, nullable=False)
-    completed_goal: Mapped[int] = mapped_column(Integer, nullable=False)
-    progress_rate: Mapped[DECIMAL] = mapped_column(DECIMAL(5, 2), default=0.00)
-
-    # CategoryProgress → User (N : 1 관계)
-    user: Mapped["User"] = relationship("User", back_populates="category_progresses")
+    category_name: Mapped[CategoryNameEnum] = mapped_column(nullable=False)
+    category_color: Mapped[CategoryColorEnum] = mapped_column(nullable=False)
 
-    # CategoryProgress → ComprehensiveEvaluation, RecommendedChallenge, Goal, MonthlyAchievement (1 : N 관계)
-    comprehensive_evaluations: Mapped[
-        List["app.models.evaluation.ComprehensiveEvaluation"]
-    ] = relationship(
-        "app.models.evaluation.ComprehensiveEvaluation",
-        back_populates="category_progress",
+    # 관계: Category 1 : 1 Progress
+    progress: Mapped["app.models.category.Progress"] = relationship(
+        "app.models.category.Progress",
+        back_populates="category",
         cascade="all, delete",
     )
+
+    # 관계: Category 1 : N Todo, Memo, Storage, RecommendedChallenge
+    todos: Mapped[Optional[List["app.models.category.Todo"]]] = relationship(
+        "app.models.category.Todo", back_populates="category", cascade="all, delete"
+    )
+    memos: Mapped[Optional[List["app.models.category.Memo"]]] = relationship(
+        "app.models.category.Memo", back_populates="category", cascade="all, delete"
+    )
+    storages: Mapped[Optional[List["app.models.chat.Storage"]]] = relationship(
+        "app.models.chat.Storage", back_populates="category", cascade="all, delete"
+    )
     recommended_challenges: Mapped[
-        List["app.models.evaluation.RecommendedChallenge"]
+        Optional[List["app.models.evaluation.RecommendedChallenge"]]
     ] = relationship(
         "app.models.evaluation.RecommendedChallenge",
-        back_populates="category_progress",
+        back_populates="category",
         cascade="all, delete",
-        foreign_keys="[app.models.evaluation.RecommendedChallenge.progress_id]",  # 추가
     )
-    goals: Mapped[List["Goal"]] = relationship(
-        "Goal", back_populates="category_progress", cascade="all, delete"
+
+
+# ✅ Progress 테이블: 카테고리별 목표 진행률 저장 테이블
+class Progress(Base):
+    __tablename__ = "progresses"
+
+    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
+    category_id: Mapped[UUID] = mapped_column(
+        ForeignKey("categories.id"), nullable=False
     )
-    monthly_achievements: Mapped[List["app.models.evaluation.MonthlyAchievement"]] = (
-        relationship(
-            "app.models.evaluation.MonthlyAchievement",
-            back_populates="category_progress",
-            cascade="all, delete",
-        )
+    progress_rate: Mapped[Optional[DECIMAL]] = mapped_column(
```

Comment:
ERD에는 not null인데 `Optional`이 붙는 이유가 궁금합니당.

```suggestion
    progress_rate: Mapped[DECIMAL] = mapped_column(
```

Reply:
우선 erd가 제데로 나오지 않아서 일단 걸어놓았고 nn으로 수정된다면 Optional은 제거하겠습니다

Reply:
@Ryan-OH 우선 ERD가 NN이라 제거해두겠습니당